src/browser-bookmarks/import-bookmarks.js

```diff
@@ -0,0 +1,41 @@
+import db from '../pouchdb'
+import { updatePageSearchIndex } from '../search/find-pages'
+import { getVisitItemsForHistoryItems, transformToPageDoc, transformToVisitDoc,
+   convertHistoryToPagesAndVisits } from '../browser-history/import-history'
+import { isWorthRemembering, generatePageDocId, generateVisitDocId,
+  visitKeyPrefix, convertVisitDocId } from '../activity-logger'
+
+    function onRejected(error) {
+      console.log(`An error: ${error}`);
+    }
+
+    function getBookmarkItems() {
+      var bookmarkTree = browser.bookmarks.getRecent(100000);
+      return bookmarkTree.then(bookmarkItems =>
+          bookmarkItems.filter(({url}) => isWorthRemembering({url})));
+    }
+
+    export default function importBookmarks() {
+      return getBookmarkItems().then(result =>
+      getVisitItemsForHistoryItems(result)).then(
+          res => {
+            var docs = convertHistoryToPagesAndVisits(res)
```

Comment:
@ Treora This does work smoothly as the bookmarks and the history hasve the same object structure. As you said , the only difference I could think of between both of them is setting the nonce according to which import uses it. To make the functions generic, I could move them to a seperate file and rename them and import them in the history and boomarks code. We could either insert a flag value into the model maybe?.